### ISSUE TITLE
plus assign and minus assign folding semantic problem

### ISSUE THREAD

**Issue by SimY4** [cheptsov/AdvancedExpressionFolding#54]
Consider the following code:
```
        List<Integer> l = new ArrayList<>();
        boolean b = l.add(1);
        
        Map<Integer, String> m = new HashMap<>();
        Optional<String> maybeVal = Optional.ofNullable(m.remove(2));
```
Its gets folded into:
```
        val l = [];
        boolean b = l += 1;
        
        val m = new HashMap<>();
        val maybeVal = Optional.ofNullable(m -= 2);
```
Here's the problem: semantically plus-assign and minus-assign should return a modified collection (or copy of a collection if original collection was immutable). But this is not the original method's semantic. 

Maybe it's better to cases where results of `Collection.add` `Map.put` `Collection.remove`, `Map.remove` is getting used further on?

**Comment by cheptsov** [comment]
Totally I agree. I will change it. It will fold only if the operation is a statement, not if it's an expression

**Comment by cheptsov** [comment]
Fixed in 0.8.1 (EAP), please check
